packages/insured-bridge-relayer/src/ProfitabilityCalculator.ts

```diff
@@ -0,0 +1,236 @@
+    // not send arrays (always will be unprofitable) and generate a warning.
+    for (const [index, priceResponse] of tokenPrices.entries()) {
+      if (priceResponse.status == "fulfilled") {
+        this.l1TokenInfo[this.l1Tokens[index]].tokenEthPrice =
+          this.l1TokenInfo[this.l1Tokens[index]].tokenType == TokenType.WETH
+            ? toBNWei("1")
+            : toBNWei(priceResponse.value[1]);
+      } else {
+        this.l1TokenInfo[this.l1Tokens[index]].tokenEthPrice = toBNWei("0");
+        this.logger.warn({
+          at: "ProfitabilityCalculator",
+          message: "Could not find token price!",
+          token: this.l1Tokens[index],
+        });
+      }
+    }
+    this.logger.debug({
+      at: "ProfitabilityCalculator",
+      message: "Updated prices",
+      tokenInfo: objectMap(this.l1TokenInfo, (value: { tokenType: TokenType; tokenEthPrice: BN }) => {
+        return { tokenType: TokenType[value.tokenType], tokenEthPrice: fromWei(value.tokenEthPrice) };
+      }),
+    });
+  }
+
+  getRelaySubmitTypeBasedOnProfitability(
+    l1Token: string,
+    cumulativeGasPrice: BN, // sum of the most recent base fee per gas and the current estimated maxPriorityFeePerGas.
+    slowRevenue: BN,
+    speedUpRevenue: BN,
+    instantRevenue: BN
+  ): RelaySubmitType {
+    this._throwIfNotInitialized();
+    if (!this.l1TokenInfo[l1Token]) throw new Error("Token info not found. Ensure to construct correctly");
+    const { tokenType, tokenEthPrice } = this.l1TokenInfo[l1Token];
+
+    this.logger.debug({
+      at: "ProfitabilityCalculator",
+      message: "Considering relay profitability",
+      l1Token,
+      tokenType: TokenType[tokenType],
+      tokenEthPrice: fromWei(tokenEthPrice),
+      cumulativeGasPrice: cumulativeGasPrice.toString(),
+      relayerDiscount: fromWei(this.relayerDiscount),
+      slowRevenue: slowRevenue.toString(),
+      speedUpRevenue: speedUpRevenue.toString(),
+      instantRevenue: instantRevenue.toString(),
+    });
+
+    // Calculate the transaction submission costs depending on the gas price and the token type being relayed. This
+    // considers the relayer discount, applied to all relay actions equally.
+    const ethSubmissionCost = this.getRelayEthSubmissionCost(cumulativeGasPrice, tokenType);
+
+    // Calculate the expected revenue, in ETH, based on the amount of tokens being offered in fees.
+    const ethRevenue = this.getEthRevenue(tokenEthPrice, slowRevenue, speedUpRevenue, instantRevenue);
+
+    // Calculate the relay profitability as the difference between revenue and cost.
+    const ethProfitability = this.getProfit(ethSubmissionCost, ethRevenue);
+
+    // Finally, determine the relay submit type based on the profitability.
+    let relaySubmitType: RelaySubmitType;
+    if (
+      ethProfitability.instantEthProfit.gt(toBN(0)) &&
+      ethProfitability.instantEthProfit.gt(ethProfitability.slowEthProfit)
```

Comment:
Did we discuss using `gte` here?

Reply:
please see here: https://github.com/UMAprotocol/protocol/pull/3656#discussion_r762763723